src/Components/Card/Card.js

```diff
@@ -1,26 +1,32 @@
 import "./Card.css";
-import { Link } from "react-router-dom";
-
+import { Link, useNavigate } from "react-router-dom";
+import CountryDetails from "../CountryDetails/CountryDetails";
 const Card = ({ name, flag, continent }) => {
+  const navigate = useNavigate();
   return (
     <div>
-      <Link to={`/${name.replace(/\s/g, "")}`}>
-        <div className=" w-40 h-52 bg-sky-blue rounded-md">
-          <div className="img-container w-35 h-25 p-2">
-            <img src={flag} className="img rounded-md h-25" />
+      {/* <Link to={`/${name.replace(/\s/g, "")}`}> */}
+      <div
+        className=" w-40 h-52 bg-sky-blue rounded-md"
+        onClick={() => {
+          navigate(`${name}`);
```

Comment:
It can simply be like navigate(name) as we are not appending any other strings with name.